src/Compilers/Core/Portable/Syntax/SyntaxNodeOrTokenList.cs

```diff
@@ -316,10 +316,10 @@ private static SyntaxNodeOrTokenList CreateList(List<SyntaxNodeOrToken> items)
                 return default(SyntaxNodeOrTokenList);
             }
 
-            var newGreen = GreenNode.CreateList(items.Select(n => n.UnderlyingNode!))!;
+            var newGreen = GreenNode.CreateList(items, static n => n.UnderlyingNode!)!;
             if (newGreen.IsToken)
             {
-                newGreen = GreenNode.CreateList(new[] { newGreen }, alwaysCreateListNode: true)!;
+                newGreen = Syntax.InternalSyntax.SyntaxList.List(new[] { newGreen });
```

Comment:
💡 Use the overload that takes `ArrayElement<GreenNode>[]` to avoid an intermediate `GreenNode[1]` allocation.